### PR TITLE
better featured search result item handling

### PR DESCRIPTION
**Pull request by akinwale**

### CODE REVIEW

File: src/component/claimList/view.js [hunk 1]
```diff
@@ -204,7 +204,12 @@ class ClaimList extends React.PureComponent {
 
     const options = this.buildClaimSearchOptions();
     const claimSearchKey = createNormalizedClaimSearchKey(options);
-    const uris = claimSearchByQuery[claimSearchKey];
+    let uris = claimSearchByQuery[claimSearchKey];
+
+    if (uris) {
+      // temporary workaround for missing short_urls
+      uris = uris.filter(uri => uri && uri.length > 0);
```
Comment (kauffj) [inline src/component/claimList/view.js]: if 0.40 fixes this, want to just make the switch?

File: src/component/fileListItem/view.js [hunk 2]
```diff
@@ -130,7 +136,7 @@ class FileListItem extends React.PureComponent {
             style={fileListStyle.thumbnail}
             duration={duration}
             resizeMode="cover"
-            title={title || name}
+            title={title || name || normalizeURI(uri).substring(7)}
```
Comment (kauffj) [inline src/component/fileListItem/view.js]: when would you have a uri but not a name for a file list item?

File: src/component/fileListItem/view.js [hunk 3]
```diff
@@ -167,6 +173,13 @@ class FileListItem extends React.PureComponent {
                 {isRewardContent && <Icon style={fileListStyle.rewardIcon} name="award" size={12} />}
               </View>
             )}
+
+            {featuredResult && !isResolving && !claim && (
+              <View style={fileListStyle.titleContainer}>
+                <Text style={fileListStyle.featuredTitle}>Nothing here. Claim this address!</Text>
```
Comment (kauffj) [inline src/component/fileListItem/view.js]: i18n

Comment (kauffj) [reply src/component/fileListItem/view.js]: Can this text be "Nothing here. Publish something!"

File: src/page/publish/view.js [hunk 4]
```diff
@@ -180,9 +181,16 @@ class PublishPage extends React.PureComponent {
 
       // Check if this is an edit action
       if (navigation.state.params) {
-        const { editMode, claimToEdit } = navigation.state.params;
+        const { editMode, claimToEdit, vanityUrl } = navigation.state.params;
         if (editMode) {
           this.prepareEdit(claimToEdit);
+        } else if (vanityUrl && vanityUrl.trim().length > 0) {
```
Comment (kauffj) [inline src/page/publish/view.js]: imo it's much better to perform this cleaning either in the reducer or the selector than to litter it in components

can you just make `vanityUrl = vanityUrl.trim()` before getting into the component at all?

File: src/component/fileListItem/view.js (hunk 2 above)
Comment (akinwale) [reply src/component/fileListItem/view.js]: This applies when there's no claim for a featured search item. The values for `title` and `name` are obtained from the claim.

**Pull request merged**